vendor/blockchain_contracts/src/bin/print_offsets.rs

```diff
@@ -1,27 +1,31 @@
-use self::calculate_offsets::{
-    ethereum::rfc003::{erc20_offsets, ether_offsets},
-    to_markdown,
-};
-
 mod calculate_offsets;
 
+use self::calculate_offsets::ethereum::rfc003::{offset::to_markdown, Error};
+use crate::calculate_offsets::ethereum::rfc003::contract::Contract;
+
 #[allow(clippy::print_stdout)]
-fn main() {
+fn main() -> Result<(), Error> {
     println!("### RFC003 ###");
 
-    {
-        println!("** Ether on Ethereum **");
-        let contract = ether_offsets::contract_template();
-        println!("Contract template:\n {}", contract);
-        let offsets = ether_offsets::get_all_offsets();
-        println!("{}", to_markdown(offsets));
-    }
-
-    {
-        println!("** ERC20 on Ethereum **");
-        let contract = erc20_offsets::contract_template();
-        println!("Contract template:\n {}", contract);
-        let offsets = erc20_offsets::all_offsets();
-        println!("{}", to_markdown(offsets));
-    }
+    print_offsets("ether")?;
+    print_offsets("erc20")?;
+
+    Ok(())
+}
+
+#[allow(clippy::print_stdout)]
+fn print_offsets(dir: &'static str) -> Result<(), Error> {
+    let contract = Contract::compile_from_directory_and_load_placeholder_config(dir)?;
+
+    let offsets = contract.calculate_offsets()?;
+    let metadata = contract.meta_data;
+
+    println!(
```

Comment:
:wrench: Put these two lines in a function `to_markdown` so that it looks like this:

```rust
println("{}", metadata.to_markdown());
println("{}", to_markdown(offsets));
```

Reply:
You might also just inline the `to_markdown` function for the offsets here, since it is actually not duplicated.

Reply:
I adapted it, did not go for inline though